wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java

```diff
@@ -797,4 +805,36 @@ private WorkspaceDto asDtoWithLinksAndToken(Workspace workspace) throws ServerEx
 
     return workspaceDto;
   }
+
+  private WorkspaceDto filterVisible(WorkspaceDto workspace, Visibility visibility) {
+    // no runtime - nothing to filter
+    if (workspace.getRuntime() == null) {
+      return workspace;
+    }
+    boolean includeInternal = Visibility.internal.equals(visibility);
+    String internalVisibilityAttribute = Visibility.internal.name();
+    for (MachineDto machine : workspace.getRuntime().getMachines().values()) {
+      HashMap<String, ServerDto> collect =
```

Comment:
shall we make this with old style loops?

Reply:
done